### PR TITLE
fix: set max decode size of proto message, add mvtx metrics

### PR DESCRIPTION
**Pull request by yhl25**

### CODE REVIEW

**Review by vigith — COMMENTED** [review]
let's also think whether (1000ms x 5) retry is enough

**Comment by yhl25** [comment]
> let's also think whether (1000ms x 5) retry is enough

I think 5 is less, I have increased the retry count to 60 (1000ms * 60).

File: rust/numaflow-core/src/shared/grpc.rs [hunk 1]
```diff
@@ -90,7 +90,7 @@ pub(crate) fn prost_timestamp_from_utc(t: DateTime<Utc>) -> Option<Timestamp> {
 
 pub(crate) async fn create_rpc_channel(socket_path: PathBuf) -> error::Result<Channel> {
     const RECONNECT_INTERVAL: u64 = 1000;
-    const MAX_RECONNECT_ATTEMPTS: usize = 5;
+    const MAX_RECONNECT_ATTEMPTS: usize = 60;
```
Comment (vigith) [inline rust/numaflow-core/src/shared/grpc.rs:93]: does this print for every retry? else users might think why it is hanging. please print a `error!` for every retry?

Comment (yhl25) [reply rust/numaflow-core/src/shared/grpc.rs:93]: I added a `warn! `between the retries, if the retries are exhausted we print `error!`.

**Review by vigith — APPROVED** [review]

**Pull request merged**